src/_pytest/python.py

```diff
@@ -1002,7 +1002,14 @@ def make_unique_parameterset_ids(self) -> List[str]:
             # Suffix non-unique IDs to make them unique.
             for index, id in enumerate(resolved_ids):
                 if id_counts[id] > 1:
-                    resolved_ids[index] = f"{id}{id_suffixes[id]}"
+                    suffix = ""
+                    if id[-1].isdigit():
+                        suffix = "_"
+                    new_id = f"{id}{suffix}{id_suffixes[id]}"
+                    while new_id in set(resolved_ids):
+                        id_suffixes[id] += 1
+                        new_id = f"{id}{suffix}{id_suffixes[id]}"
+                    resolved_ids[index] = new_id
                     id_suffixes[id] += 1
         return resolved_ids
```

Comment:
```suggestion
        assert len(resolved_ids) == len(set(resolved_ids)), f"Internal error: {resolved_ids=}"
        return resolved_ids
```
I think that the tests below would have passed before this PR too - we already had the correct _number_ of tests, just without unique IDs.  This assertion would have failed though.